specs/agents/logging.md

```diff
@@ -31,11 +30,115 @@ agents can treat that as a synonym for `error` or `fatal`.
 
 The `off` level is a switch to completely turn off logging.
 
-### Backwards compatibility
+## Backwards compatibility
 
 Most agents have already implemented `log_level`,
 accepting a different set of levels.
 Those agents should still accept their "native" log levels to preserve backwards compatibility.
 However, in central config,
 there will only be a dropdown with the levels that are consistent across agents.
 Also, the documentation should not mention the old log levels going forward.
+
+## Logging Preamble
+
+All log messages described in this section MUST be printed using the `info` logging level
+unless a different level is explicitly mentioned.
+
+The agent logging preamble consists of 3 blocks:
+
+* **Agent**: This block is mandatory and contains basic version and build date information.
+* **Environment**: This block is optional but for supportability reasons it should be provided.
+* **Configuration**: This block mandatory and contains a minimum set of relevant configuration values.
+
+### Agent
+
+On startup, all APM agents MUST log basic information regarding their technology (language, runtime),
+version information, and build date in a format chosen by the respective agent.
+
+This SHOULD be the very first log message that is created by an agent.
+
+Example:
+
+```text
+Elastic APM .NET Agent, version: 1.19.1-preview, build date: 2022-10-27 10:55:42 UTC
+```
+
+Agents SHOULD also report when they are running in a **non-production** scenario (e.g. a debug
+or pre-release build) using the `warning` logging level.
+
+Example:
+
+```text
+This a pre-release version and not intended for use in production environments!
+```
+
+### Environment
+
+Additionally, agents SHOULD report information about their environment (e.g. host, process, runtime).
+
+| Item | Description | Example |
+| - | - | - |
+| Process ID | The Process ID in decimal format. | `83606` |
+| Process Name | The executable image name or the full path to it.  | `w3wp.exe`, `/usr/local/share/dotnet/dotnet` |
+| Command Line | The full command line used to launch this process as available to the runtime. | `/Users/acme/some_app/bin/Debug/net7.0/some_app.dll foo=bar` |
+| Operating System | OS name and version in a human-readable format. | `macOS Version 12.6.1 (build 21G217)` |
+| CPU architecture | One of the well-known values from the table below. | `arm64` |
+| Host | The (optionally fully-qualified) host name. | `MacBook-Pro.localdomain` |
+| Time zone | The local time zone in UTC-offset notation. | `UTC+0200` |
+| Runtime | Name and version of the executing runtime. | `.NET Framework 4.8.4250.0`|
+
+**CPU Architecture:**
+
+| Value | Description |
+| - | - |
+| `amd64` | AMD64 |
```

Comment:
For example, Node.js' `os.arch()` returns 'x64' for the CPU architecture. Do we want to allow the language's own "spellings" of architectures, or attempt a translation to an APM common standard?

* * *

Similarly for OS name and version. On my macOS 12.6.1 machine, `node` gives me:

```
> os.platform()
'darwin'
> os.release()
'21.6.0'
```

I'd have to dig to see if there is anything straightforward to give a macOS version.
Note that a mapping table for most versions of darwin is here: https://en.wikipedia.org/wiki/Darwin_%28operating_system%29#Release_history

Reply:
I took over this from the [OpenTelemetry spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/host.md). But since this is "just" about logging and not semantic conventions/ECS we definitely can be more lenient here (I'll add note).

> ... for OS name and version. 

The "human-readable" aspect is what I was after here. There is no specific format or anything.

Reply:
I made the wording more lenient here ("... agent can decide to use different values that might ...").